addon/components/search-result/component.js

```diff
@@ -132,22 +133,35 @@ export default Ember.Component.extend(Analytics, hostAppName, {
         return false;
     }),
     hyperlink: Ember.computed('result', function() {
-        let re = null;
-        for (let i = 0; i < this.get('result.providers.length'); i++) {
-            //If the result has multiple providers, and one of them matches, use the first one found.
-            re = this.providerUrlRegex[this.get('result.providers')[i].name];
-            if (re) break;
-        }
-
-        re = re || this.providerUrlRegex.OSF;
-
+        let urlRegex = /^https?:\/\/[^\/]+\//;
+        let returnedUrl = null;
+        const domainRedirectList = this.get('domainRedirectProviders');
         const identifiers = this.get('result.identifiers').filter(ident => ident.startsWith('http://') || ident.startsWith('https://') );
-
-        for (let j = 0; j < identifiers.length; j++)
-            if (re.test(identifiers[j]))
-                return identifiers[j];
-
-        return identifiers[0];
+        for (let identifier of identifiers) {
+            // If the identifier's url matches a branded url, then use that
+            let url = identifier.match(urlRegex);
+            if (url && domainRedirectList.includes(url[0])) {
+                returnedUrl = identifier;
+            }
+        }
+        if (!returnedUrl) {
+            let re = null;
+            for (let providerVal of this.get('result.providers')) {
+                // If the result has multiple providers and one of them matches use the first one found
+                re = this.providerUrlRegex[providerVal.name];
+                if (re) {
+                    break;
+                }
```

Comment:
You can decide if you want to use curlys here or not. In the below for/if block, curlys definitely make it more readable. Here I think they are probably not needed. See https://github.com/airbnb/javascript/blob/master/README.md#blocks for details.